src/block/block_compact.c

```diff
@@ -51,6 +50,23 @@ __wt_block_compact_end(WT_SESSION_IMPL *session, WT_BLOCK *block)
     return (0);
 }
 
+/*
+ * __wt_block_compact_get_progress_stats --
+ *     Collect compact progress stats.
+ */
+void
+__wt_block_compact_get_progress_stats(WT_SESSION_IMPL *session, WT_BM *bm,
```

Comment:
Maybe we can move this method to `bt_compact.c`? It can be static in that file.

Reply:
I prefer to keep it the same way because there was a comment by Keith Bostic that it is a layering violation and btree layer should not be looking at the WT_BLOCK structure.

Reply:
Fair enough. Thanks for the explanation.

Reply:
Yes like Ravi said we generally avoid directly reading the data structures managed by other modules.